### PR TITLE
allow zero queries in Hybrid 

### PR DESCRIPTION
**Pull request by chdoc**
This PR extends `Hybrid` as follows:
- now statements using multiplication rather than division that also apply in the case where `q = 0`.
- documentation for the games that appear in the statements

In order to ensure that `HybGame` terminates even in the case where `q = 0`, the sampling is now done in `[0..max 0 (q-1)]` rather than [0..q-1].

### CODE REVIEW

**Comment by fdupress** [comment]
Thanks, @chdoc.

I'm wondering if this (breaking backwards compatibility) is not the opportunity to start switching the libraries to using locally-quantified query bounds. (So the number of queries is no longer a cloning parameter, but a parameter to the lemma.)

I'll leave it to you: since you're improving this, do you want to improve it all the way, or would you rather leave it as it is now?

**Comment by strub** [comment]
This PR is marked for the next release. I would suggest to delay any further modification to the next release.

File: theories/distributions/SDist.ec
Comment (fdupress) [inline theories/distributions/SDist.ec:658]:
```suggestion
```

File: theories/encryption/Hybrid.ec
Comment (fdupress) [inline theories/encryption/Hybrid.ec:658]: It's worth thinking about whether this lemma is generally useful in this form: We're usually trying to upper-bound the advantage of `A` in distinguishing `Ln` from `Rn`, so the multiplicative form is probably the most useful?

File: theories/encryption/PKE_hybrid.ec
Comment (fdupress) [inline theories/encryption/PKE_hybrid.ec:217]: Consider `auto=> /#.`?

**Review by fdupress — APPROVED** [review]
Good to go after consideration of minor comments.

File: theories/encryption/Hybrid.ec
Comment (chdoc) [reply theories/encryption/Hybrid.ec:658]: Yes, I agree that the multiplicative from is the one that's most useful. However, the form with division was previously the only available form, so having these lemmas eases porting.

**Comment by strub** [comment]
Please, rebase on `1.0` s.t. the checks can go on.

**Comment by chdoc** [comment]
Is this blocked by anything, or can this be merged?

**Comment by fdupress** [comment]
I think blocked mainly by the fact that we don't really have a clear maintainership structure, and most of the period since the rebase was spent dealing with other things.

I'll have a chat with @strub and merge (or he'll merge if he doesn't have time to chat).

**Pull request merged**
